lib/oanda_api/client/resource_descriptor.rb

```diff
@@ -24,10 +33,7 @@ class ResourceDescriptor
       def initialize(path, method)
         @path = path
         path.match(/\/(?<resource_name>[a-z]*)\/?(?<resource_id>\w*?)$/) do |names|
-          resource_name, resource_id = [Utils.singularize(names[:resource_name]), names[:resource_id]]
-          self.resource_klass = resource_name
-          @is_collection      = method == :get && resource_id.empty?
-          @collection_name    = Utils.pluralize(resource_name).to_sym if is_collection?
+          initialize_from_resource_name(names[:resource_name], method, resource_id: names[:resource_id])
```

Comment:
Nice! Since we're always passing in a resource_id, even if it's an empty value, prefer not to use a named argument like this.
```
initialize_from_resource_name names[:resource_name], method, names[:resource_id]
```